src/Elmish.WPF/WpfProgram.fs

```diff
@@ -6,27 +6,39 @@ open Microsoft.Extensions.Logging.Abstractions
 open Elmish
 
 
-type WpfProgram<'model, 'msg> =
+type WpfProgramT<'model, 'msg, 'viewModel> =
```

Comment:
Why change the type name?

Reply:
Since I added a type parameter on a publicly visible type, it would break backwards compatibility to only do that. By renaming the type and making a type alias on line 20 that has the exact signature as the old one, we can maintain backwards compatibility.

Reply:
Make this new type internal until it needs to be public.

Why this name?

Reply:
It follows the later pattern of using `BindingT` as a typed version of the view model property. This is the typed version of the top level program.

Reply:
I don't think the name needs to change.  If you think it needs to change, can you create a branch with code that would break?

For the same reason, we probably don't need `BindingT` either.

Reply:
Here is an example of what it would break: https://github.com/marner2/Elmish.WPF/compare/feature/make_wpfprogram_more_generic...marner2:Elmish.WPF:example/jm/wpfProgram_generic_issue?expand=1

For the same reason, we'll also need to do this for `Binding` otherwise *every* single place that explicitly specifies the `Binding` type will break.

Reply:
I think I was unclear.  That code doesn't break as I requested.

I agree that a type alias is required.  I am not convinced that the two generic types in question must have different names.

See [this branch](https://github.com/TysonMN/Elmish.WPF/tree/WpfProgram_changes) for an example of what I mean.

Reply:
Oh. I didn't realize that you could overload type names like that. Very interesting. I'll make that change.

Reply:
Changed

Reply:
Earlier in the comment thread I realized that we don't have to use the name `BindingT` either.

Have we added code using that name to `master` already?

Reply:
No. Still waiting on #511 and #512